torch/_dynamo/variables/misc.py

```diff
@@ -283,7 +283,7 @@ def _func_name(self):
         return "_C._set_grad_enabled"
 
     def fn_name(self):
-        if self.target_values:
+        if self.target_values[0]:
```

Comment:
After the refactor at https://github.com/pytorch/torchdynamo/pull/1289, ```target_values``` becomes a list, so we should get the grad state from ```[0]```. Otherwise, this line always return true.